client/src/docs-ui/page/page.tsx

```diff
@@ -169,9 +168,9 @@ export class DocsPage {
     }
   }
 
-  componentWillLoad() {
+  async componentWillLoad() {
     this.restoreBlockSwitcherState();
-    return this.getPageData();
+    return await this.getPageData();
```

Comment:
This also didn't end up doing anything (i.e. fixing the 404 header being wrong), but I think this is more correct, so I'm going to leave it.